### PR TITLE
[DOC] Tidy up developers  guide

### LINKED ISSUE #600: [DOC] Development guide needs updating
### Describe the issue linked to the documentation

the dev guide on the website
https://www.aeon-toolkit.org/en/latest/developers.html
is out of date and could do with a re-write

### Suggest a potential alternative/fix

go through check content, update where necessary

### PR DESCRIPTION
**Pull request by TonyBagnall**
resolves #600 
work through tidying up dev guide

### CODE REVIEW

File: docs/developer_guide/continuous_integration.md [hunk 1]
```diff
@@ -11,27 +8,25 @@ up your developer environment, and installed a
 `development version <https://www.aeon-toolkit.org/en/stable/installation.html>`__
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:8]: Be careful with links if you include them inside a code block. For example, this one does not create an hyperlink.

File: docs/developer_guide/continuous_integration.md [hunk 2]
```diff
@@ -11,27 +8,25 @@ up your developer environment, and installed a
...
 of aeon.
 
-Code quality checks
--------------------
+## Code quality checks
 
-.. _pre-commit: https://pre-commit.com
+``_pre-commit: <https://pre-commit.com>`
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:13]: Same here, this is shown as: 

``_pre-commit: <https://pre-commit.com>` 

and should be: 

`pre-commit`: <https://pre-commit.com>

or better:

[`pre-commit`](https://pre-commit.com)

File: docs/developer_guide/continuous_integration.md [hunk 3]
```diff
@@ -11,27 +8,25 @@ up your developer environment, and installed a
...
 
 We use `pre-commit`_ for code quality checks (a process we also refer to as "linting" checks).
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:15]: remove the "_".

File: docs/developer_guide/continuous_integration.md (hunk 1 above)
Comment (dguijo) [reply docs/developer_guide/continuous_integration.md:8]: Now it is shown as:

`development version <https://www.aeon-toolkit.org/en/stable/installation.html>`__

while it should be:

`development version` <https://www.aeon-toolkit.org/en/stable/installation.html> 

or better:

[`development version`](https://www.aeon-toolkit.org/en/stable/installation.html)

File: docs/developer_guide/continuous_integration.md [hunk 4]
```diff
@@ -11,27 +8,25 @@ up your developer environment, and installed a
...
 
-.. code:: bash
-
+```{code-block} powershell
    pre-commit install
+```
 
 pre-commit should now automatically run anything you make a commit! Please let us know if you encounter any issues getting this setup.
 
 For a detailed guide on code quality and linting for developers, see :ref:`coding_standards`.
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:27]: The :ref:`coding_standards` should be:

[`coding_standards`](https://github.com/aeon-toolkit/aeon/blob/main/docs/developer_guide/coding_standards.rst)

File: docs/developer_guide/continuous_integration.md [hunk 5]
```diff
@@ -11,27 +8,25 @@ up your developer environment, and installed a
...
 
-Unit testing
-~~~~~~~~~~~~
+## Unit testing
 
 We use `pytest <https://docs.pytest.org/en/latest/>`__ for unit testing.
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:31]: [`pytest`](https://docs.pytest.org/en/latest/) and remove the __

**Review by dguijo — CHANGES_REQUESTED** [review]
Thank you for this tedious work of changing and completing docs. 

The markdown format for the links changes a lot from rst. You can see here the [preview of one document, which is broken at the end](https://github.com/aeon-toolkit/aeon/blob/da1abff4f1c7ef6a24f2c32f6bf1d873bc3f03c1/docs/developer_guide/continuous_integration.md) 

I don't mind doing this if you want, let me know. 

Thank you again! Hard and thorough work :)

**Comment by TonyBagnall** [comment]
thanks david, the links need reformatting

**Comment by TonyBagnall** [comment]
@dguijo I've gone through catching most, if you get a chance tomorrow to check through please feel free to jump in and correct it.

File: docs/developer_guide/add_estimators.md [hunk 6]
```diff
@@ -1,71 +1,48 @@
...
-===============================================
+## Implementing an ``aeon`` compatible estimator
 
 The high-level steps to implement ``aeon`` compatible estimators are as follows:
 
 1.  identify the type of the estimator: forecaster, classifier, etc
 2.  copy the extension template for that kind of estimator to its intended location
 3.  complete the extension template
-4.  run the ``aeon`` test suite and/or the ``check_estimator`` utility (see `here <https://www.aeon-toolkit.org/en/latest/developer_guide/add_estimators.html#using-the-check-estimator-utility>`__)
+4.  run the ``aeon`` test suite and/or the ``check_estimator`` utility (see [here](https://www.aeon-toolkit.org/en/latest/developer_guide/add_estimators.html#using-the-check-estimator-utility)
```
Comment (dguijo) [inline docs/developer_guide/add_estimators.md:13]: missing ")" at the end of the link. There should be two, one for the hyperref and the other one for the (see here)

File: docs/developer_guide/coding_standards.md [hunk 7]
```diff
@@ -1,43 +1,39 @@
...
+## Coding style
 
-Coding style
-============
+We follow:
 
-In coding, we follow:
-
-*  the `PEP8 <https://www.python.org/dev/peps/pep-0008/>`__ coding guidelines. A good example can be found `here <https://gist.github.com/nateGeorge/5455d2c57fb33c1ae04706f2dc4fee01>`__.
+*  the [PEP8](https://www.python.org/dev/peps/pep-0008/>) coding guidelines. A
```
Comment (dguijo) [inline docs/developer_guide/coding_standards.md:8]: The link does not work. remove the ">".

File: docs/developer_guide/coding_standards.md [hunk 8]
```diff
@@ -1,43 +1,39 @@
...
----------------------------
+### Code formatting and linting
 
 We adhere to the following code formatting standards:
 
-* `black <https://black.readthedocs.io/en/stable/>`__ with default settings
+* [black](https://black.readthedocs.io/en/stable/) with default settings
 
-* `flake8 <https://flake8.pycqa.org/en/latest/>`__ with a ``max_line_length=88`` and some exceptions as per ``setup.cfg``
+* [flake8)(https://flake8.pycqa.org/en/latest/) with a ``max_line_length=88`` and
```
Comment (dguijo) [inline docs/developer_guide/coding_standards.md:19]: It should be [flake8](https://flake8.pycqa.org/en/latest/)

File: docs/developer_guide/coding_standards.md [hunk 9]
```diff
@@ -1,43 +1,39 @@
...
 
-This is enforced through our CI/CD workflows via `pre-commit <https://pre-commit.com/>`_.
+This is enforced through our CI/CD workflows via [pre-commit](https://pre-commit.com/)
 
 The full pre-commit configuration can be found in
-`.pre-commit-config.yaml <https://github.com/aeon-toolkit/aeon/blob/main/.pre-commit-config.yaml>`_.
+[.pre-commit-config.yaml](https://github.com/aeon-toolkit/aeon/blob/main/.pre-commit-config.yaml).
 Additional configurations can be found in
-`setup.cfg <https://github.com/aeon-toolkit/aeon/blob/main/setup.cfg>`_.
+[setup.cfg](https://github.com/aeon-toolkit/aeon/blob/main/setup.cfg).
```
Comment (dguijo) [inline docs/developer_guide/coding_standards.md:32]: can't find setup.cfg. Not sure where it is or even if it exists.

File: docs/developer_guide/coding_standards.md [hunk 10]
```diff
@@ -1,43 +1,39 @@
...
 
-``aeon`` specific code formatting conventions
------------------------------------------------
+### ``aeon`` specific code formatting conventions
 
--  Check out our :ref:`glossary`.
+-  Check out our `[glossary](https://www.aeon-toolkit.org/en/stable/glossary.html).
```
Comment (dguijo) [inline docs/developer_guide/coding_standards.md:36]: it should be:

[`glossary`](https://www.aeon-toolkit.org/en/stable/glossary.html).

File: docs/developer_guide/coding_standards.md [hunk 11]
```diff
@@ -71,29 +65,25 @@ Type the below in your python environment, and in the root of your local reposit
...
+```
 Once installed, pre-commit will automatically run all ``aeon`` code quality
 checks on the files you changed whenever you make a new commit.
 
 You can find our pre-commit configuration in
-`.pre-commit-config.yaml <https://github.com/aeon-toolkit/aeon/blob/main/.pre-commit-config.yaml>`_.
+[.pre-commit-config.yaml](https://github.com/aeon-toolkit/aeon/blob/main/.pre-commit-config.yaml>).
 Additional configurations can be found in
-`setup.cfg <https://github.com/aeon-toolkit/aeon/blob/main/setup.cfg>`_.
+[setup.cfg](https://github.com/aeon-toolkit/aeon/blob/main/setup.cfg>).
```
Comment (dguijo) [inline docs/developer_guide/coding_standards.md:83]: All this is duplicated. Consider removing. If you think it is important here as well, please, remove the setup.cfg as it does not exist.

File: docs/developer_guide/continuous_integration.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+if new pull requests do not break anything and meet code quality
+standards such as a common `coding style <#Coding-style>`__.
+Before setting up Continuous Integration, be sure that you have set
+up your developer environment, and installed a
+[development version](https://www.aeon-toolkit.org/en/stable/installation.html)
+of aeon.
+
+## Code quality checks
+
+[pre-commit:](<https://pre-commit.com)
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:13]: remove the <

File: docs/developer_guide/continuous_integration.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
+# Continuous integration
+
+We use continuous integration services on GitHub to automatically check
+if new pull requests do not break anything and meet code quality
+standards such as a common `coding style <#Coding-style>`__.
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:5]: Don't know where this ref should redirect. Also, remove the two__ at the end.

File: docs/developer_guide/continuous_integration.md [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+
+```{code-block} powershell
+    pip install -e .[dev]
+```
+
+   This installs an editable [development version](https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs>)
+   of aeon which will include the changes you make.
+
+   For trouble shooting on different operating systems, please see our detailed
+   [nstallation instructions](https://www.aeon-toolkit.org/en/latest/installation.html).
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:46]: installation

File: docs/developer_guide/continuous_integration.md [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+      pytest ./aeon
+```
+
+## Test coverage
+
+[codecov:](https://codecov.io)
+[coverage:](https://coverage.readthedocs.io/)
+[pytest-cov:](https://github.com/pytest-dev/pytest-cov)
+
+We use `coverage`_, the `pytest-cov`_ plugin, and `codecov`_ for test coverage.
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:60]: Remove the _ behind coverage, pytest-cov and codecov.

File: docs/developer_guide/continuous_integration.md [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+
+```{code-block} powershell
+      pytest ./aeon
+```
+
+## Test coverage
+
+[codecov:](https://codecov.io)
+[coverage:](https://coverage.readthedocs.io/)
+[pytest-cov:](https://github.com/pytest-dev/pytest-cov)
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:58]: Remove the ":" and introduce a new line between links. If not, they are all shown in the same line.

File: docs/developer_guide/continuous_integration.md [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+We use `coverage`_, the `pytest-cov`_ plugin, and `codecov`_ for test coverage.
+
+## Infrastructure
+
+This section gives an overview of the infrastructure and continuous
+integration services we use.
+
+| Platform      | Operation             | Configuration                       |
+|---------------|-----------------------|-------------------------------------|
+| GitHub Actions | Build/test/distribute on Linux, MacOS and Windows, run code quality checks | [`.github/workflows/`](https://github.com/aeon/skti/me/blob/main/.github/workflows/) |
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:69]: The link should be: 

https://github.com/aeon-toolkit/aeon/tree/main/.github/workflows

File: docs/developer_guide/continuous_integration.md [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+| Read the Docs | Build/deploy documentation | [`.readthedocs.yml`](https://github.com/alan-turing-institute/aeon/blob/main/.github/workflows/code-quality.yml) |
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:70]: The link should be: 

https://github.com/aeon-toolkit/aeon/blob/main/.readthedocs.yml

File: docs/developer_guide/continuous_integration.md [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+| Codecov       | Test coverage         | [`.codecov.yml`](https://github.com/aeon/aeon/blob/main/.codecov.yml), [`.coveragerc`](https://github.com/alan-turing-institute/aeon/blob/main/.coveragerc) |
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:71]: The links should be:

https://github.com/aeon-toolkit/aeon/blob/main/.codecov.yml

https://github.com/aeon-toolkit/aeon/blob/main/.coveragerc

File: docs/developer_guide/continuous_integration.md [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+
+Additional scripts used for building, unit testing and distribution can
+be found in
+[build_tools](https://github.com/aeon-toolkit/aeon/tree/main/build_tools>).
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:75]: Don't know where this folder is. Consider removing :)

File: docs/developer_guide/continuous_integration.md [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+For a detailed guide on code quality and linting for developers, see :ref:`coding_standards`.
+
+## Unit testing
+
+We use [pytest](https://docs.pytest.org/en/latest/) for unit testing.
+
+To check if your code passes all tests locally, you need to install the
+development version of aeon and all extra dependencies.
+
+1. Install the development version of aeon with developer dependencies:
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:36]: `aeon`

File: docs/developer_guide/continuous_integration.md [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+
+```{code-block} powershell
+    pip install -e .[dev]
+```
+
+   This installs an editable [development version](https://pip.pypa.io/en/stable/reference/pip_install/#editable-installs>)
```
Comment (dguijo) [inline docs/developer_guide/continuous_integration.md:42]: The link now should be: 

https://pip.pypa.io/en/stable/topics/local-project-installs/#editable-installs

File: docs/developer_guide/dependencies.md [hunk 23]
```diff
@@ -0,0 +1,43 @@
+# Dependencies
+
+## Types of dependencies
+
+There are three types of dependencies in ``aeon``: **core**, **soft**, or **developer**.
+
+.. note::
```
Comment (dguijo) [inline docs/developer_guide/dependencies.md:7]: Remove this line?

File: docs/developer_guide/dependencies.md [hunk 24]
```diff
@@ -0,0 +1,43 @@
...
+- Decorate all pytest tests that import soft dependencies with a `@pytest.mark.skipif(...)` conditional on a check to `_check_soft_dependencies` for your new soft dependency. Be sure that all soft dependencies which are imported for testing are imported within the test function itself, rather than for the whole module! This decorator will then skip your test unless the system has the required packages installed. Doing this is helpful for any users running `check_estimator` on all estimators, or a full local `pytest` run without the required soft dependency. Again, see the tests for pydarima (in forecasting) for a concrete example.
+
+## Adding a core or developer dependency
+
+Core or developer dependencies can be added only by core developers after discussion and consensus. When adding a new core dependency or changing the version of an existing one, the following files need to be updated:
+
+- `pyproject.toml`, adding the dependency or version bounds in the `dependencies` dependency set. ([pyproject.toml](https://github.com/aeon-toolkit/aeon/blob/main/pyproject.toml))
```
Comment (dguijo) [inline docs/developer_guide/dependencies.md:39]: Include the link in the first appearance of pyproject.toml

File: docs/developer_guide/dependencies.md [hunk 25]
```diff
@@ -0,0 +1,43 @@
...
+
+When adding a new developer dependency or changing the version of an existing one, the following files need to be updated:
+
+- `pyproject.toml`, adding the dependency or version bounds in the `dev` dependency set. ([pyproject.toml](https://github.com/aeon-toolkit/aeon/blob/main/pyproject.toml))
```
Comment (dguijo) [inline docs/developer_guide/dependencies.md:43]: Include the link in the first appearance of pyproject.toml

File: docs/developer_guide/documentation.md [hunk 26]
```diff
@@ -1,37 +1,28 @@
...
 
-=============
-Developing Documentation
-=============
-
-Providing instructive documentation is a key part of ``aeon's`` mission. In order to meet this,
-developers are expected to follow ``aeon's`` documentation standards.
-
-These include:
+`aeon's`` documentation standards include:
```
Comment (dguijo) [inline docs/developer_guide/documentation.md:3]: `aeon's` , remove one `

File: docs/developer_guide/documentation.md [hunk 27]
```diff
@@ -1,37 +1,28 @@
...
 
 More detailed information on ``aeon's`` documentation format is provided below.
 
-Docstring Conventions
----------------------
+## Docstring conventions
 
 aeon uses the numpydoc_ Sphinx extension and follows
-`NumPy docstring format <https://numpydoc.readthedocs.io/en/latest/format.html>`_.
+[NumPy docstring format](https://numpydoc.readthedocs.io/en/latest/format.html>).
```
Comment (dguijo) [inline docs/developer_guide/documentation.md:14]: Remove the > at the end of the link.

File: docs/developer_guide/documentation.md [hunk 28]
```diff
@@ -1,37 +1,28 @@
...
-* Adding new public functionality to the :ref:`api_reference` and :ref:`user guide <user_guide>`
+* Adding new public functionality to the [api_reference](https://www.aeon-toolkit.org/en/stable/api_reference.html) and [user_guide](https://www.aeon-toolkit.org/en/stable/getting_started.html).
 
 More detailed information on ``aeon's`` documentation format is provided below.
 
-Docstring Conventions
----------------------
+## Docstring conventions
 
 aeon uses the numpydoc_ Sphinx extension and follows
```
Comment (dguijo) [inline docs/developer_guide/documentation.md:13]: `numpydoc`
remove the _

File: docs/developer_guide/documentation.md [hunk 29]
```diff
@@ -61,57 +49,37 @@ Accordingly, aeon estimators and most other public code artifcations should gene
...
 might only list other percentage error based performance metrics.  Likewise, a distance based classifier might list other distance based classifiers but
 not include other types of time series classifiers.
 
-Notes
-~~~~~
+### Notes
 
 The notes section can include several types of information, including:
 
 - Mathematical details of a code object or other important implementation details (using ..math or :math:`` functionality)
```
Comment (dguijo) [inline docs/developer_guide/documentation.md:78]: Missing ``??

File: docs/developer_guide/testing_framework.md [hunk 30]
```diff
@@ -0,0 +1,18 @@
+# Testing framework
+
+``aeon`` uses ``pytest`` for testing interface compliance of estimators, and correctness of code.
+This page gives an overview over the tests, and introductions on how to add tests, or how to extend the testing framework.
+
+## Test module architecture
+
+``aeon`` testing happens on three layers, roughly corresponding to the inheritance layers of estimators.
+
+* "package level": testing interface compliance with the ``BaseObject`` and ``BaseEstimator`` specifications, in ``tests/test_all_estimators.py``
```
Comment (dguijo) [inline docs/developer_guide/testing_framework.md:10]: maybe better to use bold **package level** ? 

Same for **module level** and **low level**

**Review by dguijo — CHANGES_REQUESTED** [review]
What a hard job, Tony. Some of the unresolved comments that I left yesterday should be done. Thank you!

File: docs/developer_guide/documentation.md (hunk 29 above)
Comment (TonyBagnall) [reply docs/developer_guide/documentation.md:78]: happy if you want to jump in and just correct, may be quicker?

**Review by hadifawaz1999 — CHANGES_REQUESTED** [review]
Thanks @TonyBagnall , can you for update the base deep part of the diagram with the fixed version we did ? the old one says BaseDeepClusterer 3 times
basically use this one:
image

instead of this one:
image

**Review by hadifawaz1999 — APPROVED** [review]
lgtm !

**Pull request merged**